### PR TITLE
Pyzo

### PR DESCRIPTION
**Pull request by lachhebo**
Pyzo is a free and open-source computing environment based on Python.

More information here :  http://www.pyzo.org/index.html

### CODE REVIEW

File: org.pyzo.pyzo.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2014-2017 Almar Klein -->
+<!-- This is a appdata file for freedesktop software centers. -->
+<!-- Please let me know if this needs changes. -->
+<component>
+ <id type="desktop">pyzo.desktop</id>
```
Comment (TingPing) [inline org.pyzo.pyzo.appdata.xml]: Run this file under `appstream-util upgrade` for the modern format.

Comment (TingPing) [reply org.pyzo.pyzo.appdata.xml]: Also not a valid id.

File: org.pyzo.pyzo.desktop [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+TryExec=pyzo
+Exec=pyzo %f
+Categories=Development;Science;IDE;Qt;
+Keywords=Python;IDE;science;interactive;
+Icon=org.pyzo.pyzo
+Terminal=false
+StartupNotify=true
+MimeType=text/x-python;
+
+
```
Comment (TingPing) [inline org.pyzo.pyzo.desktop]: Extra whitespace, maybe harmless but probably not ideal.

File: org.pyzo.pyzo.desktop [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Name=Pyzo
+GenericName=Pyzo
+Comment=Interactive editor for scientific Python
+TryExec=pyzo
+Exec=pyzo %f
+Categories=Development;Science;IDE;Qt;
+Keywords=Python;IDE;science;interactive;
+Icon=org.pyzo.pyzo
+Terminal=false
+StartupNotify=true
```
Comment (TingPing) [inline org.pyzo.pyzo.desktop]: You'll probably need to set `StartupWMClass`

File: org.pyzo.pyzo.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
+{
+  "app-id" : "org.pyzo.pyzo",
+  "runtime" : "org.kde.Platform",
+  "runtime-version" : "5.10",
```
Comment (TingPing) [inline org.pyzo.pyzo.json]: Assuming it works you should either use 5.9 (Qt LTS) or 5.11.

File: org.pyzo.pyzo.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    {
+      "name" : "pyqt5",
+      "buildsystem" : "simple",
+      "build-commands" : [
+          "pip3 install --no-index --find-links \"file://${PWD}\" --prefix=/app PyQt5==5.9.2"
+      ],
+      "sources" : [
+          {
+              "type" : "file",
+              "url" : "https://files.pythonhosted.org/packages/3a/c6/26270f5550f00920045c2f0b222a7d03d7a64382825c68bf0bb1a51d854c/PyQt5-5.9.2-5.9.3-cp35.cp36.cp37-abi3-manylinux1_x86_64.whl",
```
Comment (TingPing) [inline org.pyzo.pyzo.json]: Avoid binaries, build from source. (Some other packages on flathub do this if you search for `pyqt`).

File: org.pyzo.pyzo.desktop [hunk 6]
```diff
@@ -8,10 +8,9 @@ TryExec=pyzo
 Exec=pyzo %f
 Categories=Development;Science;IDE;Qt;
 Keywords=Python;IDE;science;interactive;
+StartupWMClass = "pyzo"
```
Comment (TingPing) [inline org.pyzo.pyzo.desktop]: No quotes around strings, no spaces around `=`...

**Comment by nedrichards** [comment]
bot, build org.pyzo.pyzo

**Comment by nedrichards** [comment]
This builds fine, you should add [OARS](https://hughsie.github.io/oars/) and version tags to the appdata and ideally send that upstream as well. Aside from that, it's good to go.

is there anyone else from upstream who we should add to the repo?

**Comment by lachhebo** [comment]
I think you should add almar klein, he is the main developer of pyzo : 

https://github.com/almarklein

**Comment by barthalion** [comment]
Hi @almarklein, would you be interested in co-maintaining the app? One of principles of Flathub is that upstream developers should be aware of package being added here and ideally, pushing upgrades themselves.

**Comment by nedrichards** [comment]
bot, build org.pyzo.pyzo

**Comment by almarklein** [comment]
Yes, you can add me as a maintainer. I hope that @lachhebo will do most of the upgrade-pushing in practice, :) but it makes sense for me to be on the list as well.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.pyzo.pyzo

**Comment by lachhebo** [comment]
Hi, 

Can i ask why the application is still not available in flathub ?

**Comment by TingPing** [comment]
@lachhebo Please report that here: https://github.com/flathub/org.pyzo.pyzo/issues

**Comment by barthalion** [comment]
Triggered new build: https://flathub.org/builds/#/builders/2/builds/3070

It should be available now.

**Pull request closed without merging**